packages/rehype-expressive-code/src/index.ts

```diff
@@ -114,39 +165,72 @@ export type RehypeExpressiveCodeDocument = {
 	data: Record<string, unknown> | undefined
 }
 
-export type RehypeExpressiveCodeRenderer = {
-	ec: ExpressiveCode
+type RehypeExpressiveCodeEngineRenderer<T extends ExpressiveCodeEngine> = {
+	ec: T
 	baseStyles: string
 	themeStyles: string
 	jsModules: string[]
 }
 
+export type RehypeExpressiveCodeCoreRenderer = RehypeExpressiveCodeEngineRenderer<ExpressiveCodeCore>
+export type RehypeExpressiveCodeRenderer = RehypeExpressiveCodeEngineRenderer<ExpressiveCode>
```

Comment:
These wrapper types only exists for backwards compat.  Ultimately, `ec` in the base type is the instance of `ExpressiveCode/ExpressiveCodeCore` that was created but from a typing perspective, I believe it makes more sense to just have `ec: ExpressiveCodeEngine` vs the actual instance types.  If I'm understanding the code, `ExpressiveCode` class is just a tiny layer on top of the engine.  If `ExpressiveCode/ExpressiveCodeCore` classes are thought of as a `bootstrap` for the engine, they shouldn't ever have any specific API so exposing the engine directly would simplify the typings here and in a couple of other places.

Reply:
Sounds good, will leave as-is, closing this as resolved.

Reply:
Further thoughts on this at https://github.com/expressive-code/expressive-code/pull/335#discussion_r2072862226